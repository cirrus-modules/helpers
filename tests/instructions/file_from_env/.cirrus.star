load("../../../lib.star", "task", "container", "file_from_env")

def main(ctx):
    return [
        task(
            "test file_from_env",
            container(),
            env=dict(DOCKER_CONFIG="ENCRYPTED[qwerty]"),
            instructions=[
               file_from_env("DOCKER_CONFIG", "/root/.docker/config"),
            ]
        ),
        task(
            "test named file_from_env",
            container(),
            env=dict(DOCKER_CONFIG="ENCRYPTED[qwerty]"),
            instructions=[
               file_from_env(
                   name="mydocker",
                   var="DOCKER_CONFIG",
                   path="/root/.docker/config"
               )
            ]
        ),
    ]
