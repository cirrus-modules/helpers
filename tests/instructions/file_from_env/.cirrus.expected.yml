task:
  name: test file_from_env
  env:
    DOCKER_CONFIG: ENCRYPTED[qwerty]
  container:
    cpu: 2
    memory: 4096
  docker_config_file:
    path: /root/.docker/config
    variable_name: DOCKER_CONFIG
task:
  name: test named file_from_env
  env:
    DOCKER_CONFIG: ENCRYPTED[qwerty]
  container:
    cpu: 2
    memory: 4096
  mydocker_file:
    path: /root/.docker/config
    variable_name: DOCKER_CONFIG
